department-of-veterans-affairs/vets-website
# Bypass mock contact info generated by ContactInfo

## Are you removing, renaming or moving a folder in this PR?

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?

## Summary

- The `ContactInfo` component generates its own mock contact info when ran on localhost. This is not desired when the developer is using their own mocks, but there's no way to opt out of it, so this PR adds the ability to bypass generating that mock data.
- Also adds a README to the Review Information form
- I work for VA IIR, and we own this form. The change to the `ContactInfo` component though affects more than just our product.


## Related issue(s)

Closes [issue 1042](https://github.com/orgs/department-of-veterans-affairs/projects/1360/views/1?pane=issue&itemId=79507301&issue=department-of-veterans-affairs%7Cva-iir%7C1042) on the VA IIR board.

## Testing done

To verify this is working as expected:

1. Update `src/applications/personalization/profile/mocks/server.js` so that the endpoint `'GET /v0/user'` returns the `loa3UserWithNoMobilePhone` user.
2. Run the mock API so that the above change is actualized: `yarn mock-api --responses src/applications/personalization/profile/mocks/server.js`
3. Run the Review Information form: `yarn watch --env entry=welcome-va-setup-review-information`
4. Visit the form application http://localhost:3001/my-va/welcome-va-setup/review-information/contact-information
5. Confirm that the mobile phone number is empty instead of displaying 619-555-1234, as is usually generated by `ContactInfo`


## Screenshots

After these changes, you should be able to see user contact info as you have mocked it, in this case the user has no mobile phone number.

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |<img width="536" alt="Screenshot 2024-10-09 at 9 59 30 AM" src="https://github.com/user-attachments/assets/2c409df0-dcbc-4589-93d6-621f7ee48bf5">|<img width="536" alt="Screenshot 2024-10-09 at 9 59 53 AM" src="https://github.com/user-attachments/assets/2881e572-4a11-418c-b66e-3f92e3ed4be9">|
| Desktop |<img width="536" alt="Screenshot 2024-10-09 at 9 59 30 AM" src="https://github.com/user-attachments/assets/2c409df0-dcbc-4589-93d6-621f7ee48bf5">|<img width="536" alt="Screenshot 2024-10-09 at 9 59 53 AM" src="https://github.com/user-attachments/assets/2881e572-4a11-418c-b66e-3f92e3ed4be9">|





## What areas of the site does it impact?

- Impacts the Review Information form (`src/applications/personalization/review-information`).
- Impacts other VA Forms Library forms or other apps using the `ContactInfo` component (`src/platform/forms-system/src/js/components/ContactInfo.jsx`)

## Acceptance criteria

- The new property, `disableMockContactInfo`, when set to `true` results in usage of given mock data, rather than that generated in `ContactInfo`

### Quality Assurance & Testing


### Error Handling


### Authentication


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
